### PR TITLE
feat: Fragment arguments v16

### PR DESCRIPTION
**Pull request by JoviDeCroock**
This is the v16 implementation of https://github.com/graphql/graphql-js/pull/4015 and spec implementation of https://github.com/graphql/graphql-spec/pull/1081

### CODE REVIEW

**Comment by JoviDeCroock** [comment]
@github-actions publish-pr-on-npm

File: src/language/ast.ts [hunk 1]
```diff
@@ -209,11 +209,10 @@ export const QueryDocumentKeys: {
   Field: ['alias', 'name', 'arguments', 'directives', 'selectionSet'],
   Argument: ['name', 'value'],
 
-  FragmentSpread: ['name', 'directives'],
+  FragmentSpread: ['name', 'arguments', 'directives'],
   InlineFragment: ['typeCondition', 'directives', 'selectionSet'],
   FragmentDefinition: [
     'name',
-    // Note: fragment variable definitions are deprecated and will removed in v17.0.0
```
Comment (saihaj) [inline src/language/ast.ts:216]: are we removing these? I don't recall why we decided to remove these cc @yaacovCR

Comment (JoviDeCroock) [reply src/language/ast.ts:216]: Afaik it was decided because the feature wasn't being developed on

**Comment by yaacovCR** [comment]
the PR to main #4015 has been merged, with some significant changes in implementation. we can try to backport that into v16.x.x

**Comment by JoviDeCroock** [comment]
@yaacovCR I'll look into that later on when the v17 implementation is closer to solidifying

**Review by Thompson1985 — APPROVED** [review]

**Pull request closed without merging**